### PR TITLE
Refactored the project

### PR DESCRIPTION
**Pull request by ayuhsya**
- Moved all images to `/resources`
- Moved all scripts to `/scripts`
- Moved all data to `/data`
- Moved css to `/styles`
- `fuse.min.js` now loads from CDN instead of local storage

### CODE REVIEW

File: add_node.rb [hunk 1]
```diff
@@ -42,7 +42,7 @@ def default today
   ARGV.clear
 end
 
-obj = JSON.parse(File.read("data.json"))
+obj = JSON.parse(File.read("./data/data.json"))
```
Comment (athityakumar) [inline add_node.rb]: Just `obj = JSON.parse(File.read("data/data.json"))` will do.

File: add_node.rb [hunk 2]
```diff
@@ -117,10 +117,10 @@ def default today
 
 if not testing
   puts "#{obj.length} papers now."
-  File.delete("data.json")
+  File.delete("./data/data.json")
```
Comment (athityakumar) [inline add_node.rb]: Just `File.delete("data/data.json")` will do.

File: add_node.rb [hunk 3]
```diff
@@ -117,10 +117,10 @@ def default today
...
   if pretty
-    File.open("data.json", "w") { |file| file.write(JSON.pretty_generate(obj)) }
+    File.open("./data/data.json", "w") { |file| file.write(JSON.pretty_generate(obj)) }
```
Comment (athityakumar) [inline add_node.rb]: Same issue as above comment.

File: add_node.rb [hunk 4]
```diff
@@ -117,10 +117,10 @@ def default today
...
   else
-    File.open("data.json", "w") { |file| file.write(JSON.generate(obj)) }
+    File.open("./data/data.json", "w") { |file| file.write(JSON.generate(obj)) }
```
Comment (athityakumar) [inline add_node.rb]: Same issue as above comment.

**Review by athityakumar — CHANGES_REQUESTED** [review]
Hey @ayuhsya ! Thanks for opening this PR! :smile: 

Have a look at the changes suggested. Also, typically, the sitemap looks like this - 

```
resources/
     images/
     js/ or scripts/
     css/ or styles/
```

Change the html files accordingly, to fit this the above sitemap.

**Comment by athityakumar** [comment]
@ayuhsya - No need to move the `data/` folder into `resources/`. Just make this change, and I'll merge it, if no other changes are required.

File: index.html [hunk 5]
```diff
@@ -1,10 +1,11 @@
 <html>
     <head>
         <script src="https://code.jquery.com/jquery-2.1.4.min.js"></script>
-        <script src="string_score.min.js"></script>
+        <script src="./resources/scripts/string_score.min.js"></script>
```
Comment (athityakumar) [inline index.html]: Change all the `./resources/` in this file to `resources/`. And we're good to go. :+1:

File: index.html [hunk 6]
```diff
@@ -80,9 +91,9 @@
 
              $('#query').focus();
 
-             var worker = new Worker('worker2.js');
+             var worker = new Worker('./resources/scripts/worker2.js');
 
-             $.getJSON("data.json")
+             $.getJSON("./data/data.json")
```
Comment (athityakumar) [inline index.html]: Change this to `$.getJSON("data/data.json")`

File: resources/scripts/worker.js [hunk 7]
```diff
@@ -1,5 +1,3 @@
-importScripts('fuse.min.js');
```
Comment (athityakumar) [inline resources/scripts/worker.js]: I don't think that the `new Fuse()` function in this file will work without import `fuse.min.js` file. We do need it.

Comment (ayuhsya) [reply resources/scripts/worker.js]: @athityakumar How can I test it?

Comment (athityakumar) [reply resources/scripts/worker.js]: For usual JS testing, you can check out the `Console` on your browser (Inspect > Console option appears). However, testing for worker js files would have to be done on server - you can push to your fork, publish that branch in your fork as a `gh-pages` website. Then, have a look at ayuhsya.github.io/mfqp/ and check the Console of that website.

**Comment by amrav** [comment]
Hmm, why can't you test js workers in the browser locally?

File: resources/scripts/worker.js (hunk 7 above)
Comment (athityakumar) [reply resources/scripts/worker.js]: @ayuhsya - Or you can just restore the `fuse.min.js` file back, and I'll merge this. Maybe we can work on loading `fuse.min.js` from CDN on another PR. :smile:

**Comment by icyflame** [comment]
@athityakumar @ayuhsya I think you guys don't know about `SimpleHTTPServer` :slightly_smiling_face: 

You can test everything (including JavaScript) by running: 

`$ python -m SimpleHTTPServer` 

from the `mfqp/` directory and then visiting `localhost:8000` on your browser.

@amrav This should fix their problem.

**Comment by ayuhsya** [comment]
@icyflame I was using `SimpleHTTPServer` itself to test the HTML all this while, there's some Apache thing to create local server too, but I find it inconvenient to use. 

I just can't find any usage of the file `worker.js` in the project that's why I asked how to test it. I'll just restore the `fuse` script for this PR.

**Review by athityakumar — APPROVED** [review]

**Comment by amrav** [comment]
You can also use node's http-server, which is async and gives slightly
nicer output IMO. (npm install -g http-server)

I think worker.js is unused. This whole thing was hacked together in a
night or two, so code quality went out of the window - please
reactor/rewrite liberally :)

**Comment by icyflame** [comment]
@athityakumar refactor looks good. repo looks pretty now. Thanks! :smile:

**Pull request merged**